### PR TITLE
kokkos: modify standlone test to run with +rocm

### PR DESCRIPTION
**Pull request by afzpatel**
Setting CMAKE_PREFIX_PATH with the ROCm packages for kokos+rocm standalone test

### CODE REVIEW

File: var/spack/repos/builtin/packages/kokkos/package.py [hunk 1]
```diff
@@ -416,8 +416,20 @@ def test_run(self):
...
+        if self.spec.satisfies("+rocm"):
+            rocm_prefixes = ";".join(
+                [
+                    self.spec["rocprim"].prefix,
+                    self.spec["hip"].prefix,
+                    self.spec["llvm-amdgpu"].prefix,
+                    self.spec["comgr"].prefix,
+                    self.spec["hsa-rocr-dev"].prefix,
+                ]
+            )
```
Comment (cedricchevalier19) [inline var/spack/repos/builtin/packages/kokkos/package.py]: It looks weird that we have to add all these paths explicitly.

**Review by cedricchevalier19 — CHANGES_REQUESTED** [review]
I do not think this is a good way to do it.

`spack` probably has some ways to make the `cmake()` function inherit the configuration of the package, where `rocm` is correctly found.

@spack/merge-to-develop : Can some spack gurus give some advice here?

@afzpatel: Can you add some contexts as to why you need this PR?

**Comment by afzpatel** [comment]
> @afzpatel: Can you add some contexts as to why you need this PR?

I'm trying to run the standalone kokkos test(`spack test run kokkos`) on the +rocm variant of kokkos and I get this error:
 ```
 CMake Error at /dockerx/spack/opt/spack/linux-ubuntu22.04-skylake_avx512/gcc-11.4.0/cmake-3.30.2-p4bgm3sa3x77w6z3l3xztmufc4cdmttl/share/cmake-3.30/Modules/CMakeFindDependencyMacro.cmake:76 (find_package):
  By not providing "Findrocprim.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "rocprim", but
  CMake did not find one.

  Could not find a package configuration file provided by "rocprim" with any
  of the following names:

    rocprimConfig.cmake
    rocprim-config.cmake

  Add the installation prefix of "rocprim" to CMAKE_PREFIX_PATH or set
  "rocprim_DIR" to a directory containing one of the above files.  If
  "rocprim" provides a separate development package or SDK, be sure it has
  been installed.
Call Stack (most recent call first):
  /dockerx/spack/opt/spack/linux-ubuntu22.04-skylake_avx512/gcc-11.4.0/rocthrust-6.2.0-phqn5pbivh4cw2wgibvhy27ka6c642m6/lib/cmake/rocthrust/rocthrust-config.cmake:90 (find_dependency)
  /dockerx/spack/opt/spack/linux-ubuntu22.04-skylake_avx512/gcc-11.4.0/cmake-3.30.2-p4bgm3sa3x77w6z3l3xztmufc4cdmttl/share/cmake-3.30/Modules/CMakeFindDependencyMacro.cmake:76 (find_package)
  /dockerx/spack/opt/spack/linux-ubuntu22.04-skylake_avx512/gcc-11.4.0/kokkos-4.4.00-xyou34cqnkgg5fmxh2iw3spbfiwvyrec/lib/cmake/Kokkos/KokkosConfig.cmake:42 (FIND_DEPENDENCY)
  CMakeLists.txt:3 (find_package)
 ```

File: var/spack/repos/builtin/packages/kokkos/package.py (hunk 1 above)
Comment (tldahlgren) [reply var/spack/repos/builtin/packages/kokkos/package.py]: This is how other packages have solved the problem (e.g., `hip`, `slate`).

I am unable to build the package at this time.  IF `CMAKE_PREFIX_PATH` is available and already contains all the necessary prefixes, an alternative I've seen is to use `os.environ["CMAKE_PREFIX_PATH"].replace(":", ";")`.

Comment (tldahlgren) [reply var/spack/repos/builtin/packages/kokkos/package.py]: 🤔 I see `llvm` uses `spack.build_environment.get_cmake_prefix_path(self)` to get the paths during a post-install method.  Not something I've tried yet but might have potential.

Comment (afzpatel) [reply var/spack/repos/builtin/packages/kokkos/package.py]: I changed it to use `spack.build_environment.get_cmake_prefix_path(self)`. Thanks for the suggestion!

Comment (tldahlgren) [reply var/spack/repos/builtin/packages/kokkos/package.py]: And it works for the stand-alone test?  If so, that's great!

File: var/spack/repos/builtin/packages/kokkos/package.py [hunk 2]
```diff
@@ -416,8 +416,12 @@ def test_run(self):
             raise SkipTest(f"{cmake_path} is missing")
 
         cmake = self.spec["cmake"].command
-        cmake(cmake_path, "-DEXECUTABLE_OUTPUT_PATH=" + cmake_path)
+        cmake_args = ["-DEXECUTABLE_OUTPUT_PATH=" + cmake_path]
+        if self.spec.satisfies("+rocm"):
+            prefix_paths = ";".join(spack.build_environment.get_cmake_prefix_path(self))
+            cmake_args.append("-DCMAKE_PREFIX_PATH={0}".format(prefix_paths))
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/kokkos/package.py:422]: FYI. We are encouraging maintainers to convert from string `format()` to f-strings (https://docs.python.org/3/tutorial/inputoutput.html) as we do the same in the core.

**Review by tldahlgren — APPROVED** [review]

**Review by bernhardkaindl — APPROVED** [review]

**Pull request merged**